lib/std/chanutil/chan.go

```diff
@@ -0,0 +1,47 @@
+package chanutil
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+var ErrChannelClosed = errors.New("context canceled")
```

Comment:
I don't think we should be defining our own errors that mask over well-known messages from the standard library.

This will be super confusing if we see "context canceled" and it didn't originate from the actual Golang "context" package . Same for "deadline exceeded"

I think if we want to introduce our own errors, we should be using distinct types to allow error checking more easily. Much like k8s does:

```go
type errChannelClosed struct {}

func (e errChannelClosed) Error() string {
  return "channel closed"
}

func IsChannelClosed(e error) bool {
  if _, ok := e.(errChannelClosed); ok {
    return true
  }
  return false
}
```

Reply:
This is a typo, this should be "channel closed" not "context conceled".

I'm not opposed to using types for our own errors and I was a die hard fan of this previously, but defining errors like this that don't change is pretty engrained in the standard library and they've extended support of the errors package to check for this exact type of error. I've come to accept that this is idomatic go to define errors like this and it's not going away.

Reply:
Yeah, I don't feel compelled to follow this pattern just because it's in the standard library.

I'm fine to leave this as is, but wanted to make a philosophical point here. The stdlib does plenty of other non-idiomatic things - it's not the final authority on what Good Golang is.

We're writing our own library because of deficiencies in the standard library. We should get to choose what that library looks like.